src/genie/libs/parser/iosxr/show_rib.py

```diff
@@ -0,0 +1,200 @@
+                result_dict[table_id].update({'safi': group['safi']})
+                result_dict[table_id].update({'table_id': group['table_id']})
+                result_dict[table_id].update({'prefix_limit': int(group['prfx_lmt'])})
+                result_dict[table_id].update({'prefix_count': int(group['prfx_cnt'])})
+                result_dict[table_id].update({'table_version': int(group['tbl_ver'])})
+                
+                if group['pfx_notif']=='N':
+                    result_dict[table_id].update({'prefix_limit_notified': 'No'})
+                elif group['pfx_notif']=='Y':
+                    result_dict[table_id].update({'prefix_limit_notified': 'Yes'})
+                    
+                if group['forw_refe']=='N':
+                    result_dict[table_id].update({'forward_referenced': 'No'})
+                elif group['forw_refe']=='Y':
+                    result_dict[table_id].update({'forward_referenced': 'Yes'})
+                    
+                if group['tbl_del']=='N':
+                    result_dict[table_id].update({'table_deleted': 'No'})
+                elif group['tbl_del']=='Y':
+                    result_dict[table_id].update({'table_deleted': 'Yes'})
+
+                if group['tbl_conv']=='N':
+                    result_dict[table_id].update({'table_reached_convergence': 'No'})
+                elif group['tbl_conv']=='Y':
+                    result_dict[table_id].update({'table_reached_convergence': 'Yes'})                                
+        
+        result_dict = {'table_id': result_dict}
+        result_dict = {'rib_tables': result_dict}
+        return result_dict
+
+
+# =============================================
+# Parser for 'show rib tables summary'
+# =============================================
+
+class ShowRibTablesSummarySchema(MetaParser):
+    
+    schema = {'rib_summary': {
+                        'table_id':{
+                            
+                         Any(): {
+                            'num_unicast_tables': int,
+                            'total_unicast_prefixes': int,
+                            'num_multicast_tables': int,
+                            'total_multicast_prefixes': int,
+                        }
+                    }
+                }
+            }        
+
+
+
+class ShowRibTablesSummary(ShowRibTablesSummarySchema):
+    """ Parser for show rib tables summary"""
+
+    cli_command = ['show rib tables summary']
+
+    def cli(self,port="",output=None):
+        """parsing mechanism: cli
+        Function cli() defines the cli type output parsing mechanism which
+        typically contains 3 steps: exe
+        cuting, transforming, returning
+        """
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+            
+        result_dict = {}
+        
+        p1 = re.compile(r'^Summary\sof.+in +(?P<rib>\S+)')
```

Comment:
Example for the regexp to help maintain the code in the future

Reply:
Example has been added in the comments